### PR TITLE
Add support for setting actions permissions on a repository

### PR DESCRIPTION
**Pull request by h2floh**
👋  Hello there!

Couldn't find a PR for it yet so added functionality requested in https://github.com/google/go-github/issues/2297
Sorry if I pushed myself before @cfarrend in doing it but I will need the changes.

Went through the [contributing guide](https://github.com/google/go-github/blob/master/CONTRIBUTING.md) and also added an example to test the change end2end.

Thanks everyone!

Fixes: #2297.

### CODE REVIEW

**Comment by h2floh** [comment]
Hey @gmlewis , I signed the ICLA after my company asked me to do that. Can you please rerun the cla check. Thanks!

File: github/repos_actions_allowed.go
Comment (gmlewis) [inline github/repos_actions_allowed.go:44]: Let's please follow the same pattern here of explicit err and success paths:

```
	if err != nil {
		return nil, resp, err
	}

	return p, resp, nil
```

File: github/repos_actions_permissions.go
Comment (gmlewis) [inline github/repos_actions_permissions.go:13]:
```suggestion
// ActionsPermissionsRepository represents a policy for repositories and allowed actions in a repository.
```

File: github/repos_actions_permissions.go
Comment (gmlewis) [inline github/repos_actions_permissions.go:55]: Please add a blank line before 55.

File: github/repos_actions_permissions.go
Comment (gmlewis) [inline github/repos_actions_permissions.go:57]: Let's please follow the same pattern here of explicit err and success paths:

```
	if err != nil {
		return nil, resp, err
	}

	return permissions, resp, nil
```

File: github/repos_actions_permissions.go
Comment (gmlewis) [inline github/repos_actions_permissions.go:49]:
```suggestion
func (s *RepositoriesService) EditActionsPermissions(ctx context.Context, owner, repo string, actionsPermissionsRepository ActionsPermissionsRepository) (*ActionsPermissionsRepository, *Response, error) {
```

File: github/repos_actions_permissions.go
Comment (gmlewis) [inline github/repos_actions_permissions.go:29]:
```suggestion
func (s *RepositoriesService) GetActionsPermissions(ctx context.Context, owner, repo string) (*ActionsPermissionsRepository, *Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @h2floh !
Just a few tweaks, please, then we should be ready for a second review before merging.

File: github/repos_actions_allowed.go
Comment (h2floh) [reply github/repos_actions_allowed.go:16]: I guess this syntax means that org and repo are string? Interesting Till I Learn

File: github/repos_actions_allowed.go
Comment (h2floh) [reply github/repos_actions_allowed.go:44]: Ah, ok now I finally begin to understand that pattern 👍 

Just a remark: I just copied from the existing [org level code](https://github.com/h2floh/go-github/blob/24a732e7e42d188235682fedf979c3a0ff49e1d6/github/orgs_actions_permissions.go#L57).

Comment (gmlewis) [reply github/repos_actions_allowed.go]: > Just a remark: I just copied from the existing [org level code](https://github.com/h2floh/go-github/blob/24a732e7e42d188235682fedf979c3a0ff49e1d6/github/orgs_actions_permissions.go#L57).

Thank you, @h2floh !  Please take a look at #2318 when you get a chance and LGTM+Approve if you agree with the changes.

File: github/repos_actions_allowed_test.go
Comment (gmlewis) [inline github/repos_actions_allowed_test.go:85]: Please add:

```
	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
		got, resp, err := client.Repositories.EditActionsAllowed(ctx, "o", "r", *input)
		if got != nil {
			t.Errorf("testNewRequestAndDoFailure %v = %#v, want nil", methodName, got)
		}
		return resp, err
	})
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/repos_actions_allowed_test.go
Comment (h2floh) [reply github/repos_actions_allowed_test.go:93]: Done, nice catch

File: github/repos_actions_permissions_test.go
Comment (h2floh) [reply github/repos_actions_permissions_test.go:94]: Done, nice catch

**Comment by gmlewis** [comment]
@Parker77 - do you have time for a code review?

**Review by gmlewis — APPROVED** [review]
Thank you, @h2floh !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Comment by gmlewis** [comment]
Thank you, @Parker77 !
Merging.

**Pull request merged**
